src/topology/instances/add_circle.lean

```diff
@@ -286,3 +286,115 @@ local attribute [instance] fact_zero_lt_one
 /-- The unit circle `ℝ ⧸ ℤ`. -/
 @[derive [compact_space, normal_space, second_countable_topology]]
 abbreviation unit_add_circle := add_circle (1 : ℝ)
+
+
+section identify_Icc_ends
+/-! This section proves that the natural map from `[0, T] ⊂ ℝ` to `add_circle T` gives an
+identification of `add_circle T`, as a topological space, with the quotient of `[0, T]` by the
+equivalence relation identifying `0` with `T`. -/
+
+variables (T : ℝ) [hT : fact (0 < T)]
```

Comment:
Can you generalize this to
```suggestion
variables {𝕜 : Type u_1} [linear_ordered_add_comm_group 𝕜] [topological_space 𝕜] [order_topology 𝕜] (T : 𝕜) [hT : fact (0 < T)]
```

Reply:
This would be a great deal more work, because the current argument relies on `Icc 0 T` being compact in order to show continuity of the inverse. Generalising to cover (for instance) `𝕜 = ℚ` would need a rather different, and I suspect much longer, argument; and I have grave doubts about whether this case is likely to be interesting or useful for anything.

Reply:
> This would be a great deal more work, because the current argument relies on `Icc 0 T` being compact in order to show continuity of the inverse.

Then why not use `{𝕜 : Type u_1} [linear_ordered_add_comm_group 𝕜] [topological_space 𝕜] [order_topology 𝕜]` until you reach the lemma about continuity, and switch to `ℝ` for that statement? Almost all the previous statements hold in the general case, right?

Reply:
I couldn't get it to work in that generality; but the non-continuity-related statements go through OK under the slightly more restrictive conditions
`[linear_ordered_field 𝕜] [topological_space 𝕜] [order_topology 𝕜] [archimedean  𝕜]`. Would that be a reasonable compromise?

Reply:
Sure, that looks good enough for now.